platform/engineering/request-for-comment/tracking-submissions-from-origin-to-destination.md

```diff
@@ -0,0 +1,399 @@
+
+In those cases, we could add an optional [salt](https://en.wikipedia.org/wiki/Salt_(cryptography)) value to the context. This could be used if set, but allowed to be empty when no salt is needed.
+
+### Context Changes
+
+If, through the process of submission, values used in the context of the id change, the resulting id would also change.
+
+**Examples**:
+
+#### Format Changes
+
+- Veteran submits evidence as a PNG file to VA.org.
+- That evidence is sent to Lighthouse where it is converted to a PDF file and send to CMP.
+
+If the Submission ID is generated using a hash of the PNG file as context, that context would be lost in the conversion to PDF.
+
+#### Context Loss
+
+- Veteran submits data for a form that is too long to fit in the resulting form area
+- When the PDF form is generated, some of the data is cut off or moved to an alternate section
+
+If the Submission ID is generated using that full data, the context will have been changed thus changing the ID.
+
+## Alternatives
+
+### Random IDs
+
+Assign a random unique ID for each submission. Assignment is much easier at origin as a properly unique value, such as a UUID, can generated without any context needed.
+
+A downside would be that the ID would need to either be included in the submission content or be purposely carried by each system the submission travels through in the metadata of the requests.
+
+### Central ID Map
+
+Each system in the submission chain generates its own unique ID. Once a submission has moved from SystemA to SystemB, SystemA will report to a central location the corelation of SystemA.id to SystemB.id.
+A chain would be formed in the central system such that `SystemA.id -> SystemB.id -> ... -> SystemN.id`.
+
+A downside to this method is that a failure to report a link at any step would result in the loss of visibility in all of the following steps until the missing step is found. Until the link is recovered, you cannot assume the submission failed as the system could have successfully handed the submission over but failed to notify the central system.
+
+## Additions
+
+There are some additional items and thoughts.
+
+### Origin
+
+One thought is the option to add the origin to the id, either as a concatenation of an origin id and the hash or including the origin in the hash.
+
+### ID Encoding
+
+The hash value could be representing in many ways. Hex is common, however there are options to make it shorter using Base64 or Base58.
+
+|  Type  | Value                                                              |
+| :----: | :----------------------------------------------------------------- |
+|  Hex   | `d0b0072b8286f9cf8803dfce075be3e6f255d2f9ec30e48ca9cbc872ae3cf7f6` |
+| Base58 | `F3dUukF8vdguQUp5VHurBBC4m7aTzdtgunmL4HmMSvBP`                     |
+| Base64 | `0LAHK4KG+c+IA9/OB1vj5vJV0vnsMOSMqcvIcq489/Y=`                     |
+
+Base58 is preferred as there are less special characters, making it easier to read.
```

Comment:
While I appreciate the ability to eyeball the id, I would not want that to be a deciding factor.